File: mobile/app/lib/api/models/auth_response.dart
import './oauth_token.dart';
import './user.dart';

class AuthResponse {
  final OAuthToken tokens;
  final User user;

  AuthResponse({
    required this.tokens,
    required this.user,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      tokens: OAuthToken.fromJson(json['tokens']),
      user: User.fromJson(json['user']),
    );
  }
}

File: backend/app/schemas/auth_response.py
from pydantic import BaseModel
from app.schemas.user_response import UserResponse
from app.schemas.oauth_token import OAuthToken


class AuthResponse(BaseModel):
    tokens: OAuthToken
    user: UserResponse

    class Config:
        model_config = {
        "from_attributes": True  # <-- allows conversion from ORM objects
    }
